syntax = "proto3";

package Lazysplits.Proto;

enum TargetType{
	TGT_UNUSED = 0;
	TGT_RESET = 1;
	TGT_START = 2;
	TGT_STANDARD = 3;
	TGT_END = 4;
}
enum WatchType{
	WT_UNUSED = 0;
	WT_COLOR = 1;
	WT_IMAGE_STATIC = 2;
	WT_TEXT = 3;
}
message UScalar2d{
	uint32 x = 1;
	uint32 y = 2;
}
message RGBColor{
	uint32 R = 1;
	uint32 G = 2;
	uint32 B = 3;
}

message GameList{
	message GameEntry{
		string name = 1;
		string relative_path = 2;
	}

	repeated GameEntry games = 1;
}

message GameInfo{
	message TargetEntry{
		string name = 1;
		string relative_path = 2;
	}

	string name = 1;
	repeated TargetEntry targets = 2;
}

message TargetInfo{
	message WatchEntry{
		string name = 1;
		string relative_path = 2;
		uint32 index = 3;
	}

	string name = 1;
	TargetType type = 2;
	sint32 split_offset_ms = 3;
	repeated WatchEntry watches = 4;
}

message WatchInfo{
	message WatchArea{
		UScalar2d loc = 1;
		UScalar2d size = 2;
	}

	string name = 1;
	WatchType type = 2;
	UScalar2d base_dimensions = 3;
	WatchArea area = 4;

	RGBColor color = 5;
	string img_relative_path = 6;
}